File: app/src/main/java/com/erkaslan/moviedb/ui/SharedViewModel.java
package com.erkaslan.moviedb.ui;

import android.app.Application;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.MutableLiveData;

import com.erkaslan.moviedb.MovieInterface;
import com.erkaslan.moviedb.Repository;
import com.erkaslan.moviedb.SingleMovieInterface;
import com.erkaslan.moviedb.data.Movie;

import java.util.ArrayList;

public class SharedViewModel extends AndroidViewModel {

    MutableLiveData<ArrayList<Movie>> moviesAll = new MutableLiveData<ArrayList<Movie>>();
    MutableLiveData<Movie> singleMovie = new MutableLiveData<Movie>();
    private static final String TAG = "HomeVM";
    private Repository repository;


    public SharedViewModel(@NonNull Application application) {
        super(application);
        repository = new Repository(application);

    }

    public MutableLiveData<ArrayList<Movie>> getMovieList(){
        return moviesAll;
    }

    public void setMovieList(String search) {

        repository.setMovieList(search, new MovieInterface() {
            @Override
            public void onSuccess(ArrayList<Movie> movieList) {
                //return movieList;
                Log.v(TAG, "****LIST in ONSuccess");
                Log.v(TAG, "****LIST " + String.valueOf(movieList.size()));
                moviesAll.postValue(movieList);
            }
        });
    }

    public MutableLiveData<Movie> getSingleMovie() {return singleMovie;}

    public void setSingleMovie(String imdbID) {

        repository.setSingleMovie(imdbID, new SingleMovieInterface() {
            @Override
            public void onSingleSuccess(Movie movie) {
                Log.v(TAG, "****LIST in ONSuccess");
                singleMovie.postValue(movie);
            }
        });
    }

}

File: app/src/main/java/com/erkaslan/moviedb/ui/SharedViewModel.kt
package com.erkaslan.moviedb.ui

import android.app.Application
import android.util.Log
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.erkaslan.moviedb.data.Movie
import com.erkaslan.moviedb.data.MovieInterface
import com.erkaslan.moviedb.data.Repository
import com.erkaslan.moviedb.data.SingleMovieInterface
import java.util.ArrayList

class SharedViewModel(application: Application) : AndroidViewModel(application) {
    private var _moviesAll: MutableLiveData<ArrayList<Movie>?> = MutableLiveData()

    val moviesAll: MutableLiveData<ArrayList<Movie>?>
            get() = _moviesAll

    private var _singleMovie: MutableLiveData<Movie?> = MutableLiveData()

    val singleMovie: MutableLiveData<Movie?>
        get() = _singleMovie

    private val repository: Repository

    fun setMovieList(search: String?) {
        repository.setMovieList(search!!, object : MovieInterface {
            override fun onSuccess(movieList: ArrayList<Movie>) {
                Log.v(TAG, "****LIST " + movieList!!.size.toString())
                moviesAll.value = movieList
            }
        })
    }

    fun setSingleMovie(imdbID: String?) {
        repository.setSingleMovie(imdbID!!, object : SingleMovieInterface {
            override fun onSingleSuccess(movie: Movie) {
                Log.v(TAG, "****LIST in OnSuccess")
                singleMovie.value = movie
            }
        })
    }

    companion object {
        private const val TAG = "HomeVM"
    }

    init {
        repository = Repository(application)
    }
}